src/coreclr/jit/hwintrinsic.cpp

```diff
@@ -561,7 +561,10 @@ GenTree* Compiler::getArgForHWIntrinsic(var_types            argType,
         arg = impPopStack().val;
         assert(varTypeIsArithmetic(arg->TypeGet()) || ((argType == TYP_BYREF) && arg->TypeIs(TYP_BYREF)));
 
-        assert(genActualType(arg->gtType) == genActualType(argType));
+        if (!impCheckImplicitArgumentCoercion(argType, arg->gtType))
+        {
+            BADCODE("the hwintrinsic argument has a type that can't be implicitly converted to the signature type");
+        }
```

Comment:
Ideally we should have this check outside the `else` but let's not rerun CI for it